kentikapi/client.go

```diff
@@ -78,26 +75,67 @@ type Config struct {
 	// A Timeout of zero means no timeout.
 	// Default: 100 seconds.
 	Timeout *time.Duration
-	// DisableTLS disables TLS for client connections.
-	// It has effect only on gRPC services: CloudExportAdmin, SyntheticsAdmin and SyntheticsData.
-	DisableTLS bool
 }
 
 type RetryConfig = httputil.RetryConfig
 
+type ConfigOption func(*Config)
+
+func WithAPIURL(apiURL string) ConfigOption {
+	return func(c *Config) {
+		c.APIURL = apiURL
+	}
+}
+
+func WithAuthEmail(authEmail string) ConfigOption {
+	return func(c *Config) {
+		c.AuthEmail = authEmail
+	}
+}
+
+func WithAuthToken(authToken string) ConfigOption {
+	return func(c *Config) {
+		c.AuthToken = authToken
+	}
+}
+
+func WithTimeout(timeout *time.Duration) ConfigOption {
+	return func(c *Config) {
+		if timeout != nil {
+			c.Timeout = timeout
+		}
+	}
+}
+
+func WithRetryConfig(retryCfg RetryConfig) ConfigOption {
+	return func(c *Config) {
+		retryCfg.FillDefaults()
+		c.RetryCfg = retryCfg
+	}
+}
+
+func WithLogPayloads() ConfigOption {
+	return func(c *Config) {
+		c.LogPayloads = true
+	}
+}
+
 // NewClient creates a new Kentik API client.
-func NewClient(c Config) (*Client, error) {
+func NewClient(opts ...ConfigOption) (*Client, error) {
+	c := Config{}
```

Comment:
Please just initialize it here with default values (or call config constructor). No need to split it to two steps now.

Reply:
Done